### PR TITLE
Experiment Nathan interaction system update

### PR DESCRIPTION
**Pull request by nate3323**
The interaction system now has the inventory system as  a core part of the cinematic interaction. The Island Heart can now except all versions of the JellyDewIdentity items, and Jellies respond to all versions of food items. This should allow us to modify how they respond to each item type as necessary. I have also removed the feed buttons as we can now use the drag and drop system when interacting during the cinematic interactions. The island heart can also passively pick up dropped Jelly dew and a dropped generator, although this feature should be changed in the future as the collision detection system will most likely not be the best path moving forward.

### CODE REVIEW

File: Assets/Scripts/InteractionSystem/InteractableWithUIMode.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class InteractableWithUIMode : Interactable
+public abstract class InteractableWithUIMode : Interactable
```
Comment (CaseyWalshWarder) [inline Assets/Scripts/InteractionSystem/InteractableWithUIMode.cs]: The chest and cauldron prefabs use this script, so it shouldn't be abstract.

**Review by CaseyWalshWarder — COMMENTED** [review]
Took a quick look since PRs are building up.

**Comment by nate3323** [comment]
Switched it to a virtual method for now, although I might recommend it eventually be made into an abstract class

**Comment by CaseyWalshWarder** [comment]
Looks good. The code is improved, and the UI is better when interacting with things. In the future, we should probably make it so the inventory is always open when interacting. For example, if you toggle it on and then start interacting, it toggles off. We should also probably get rid of the OnTriggerEnter in IslandHeartLeveling, but we might need it depending on how the tutorials are designed.

**Pull request merged**